tests/voq/test_fabric_cli_and_db.py

```diff
@@ -172,122 +177,127 @@ def test_fabric_cli_isolate_supervisor(duthosts, enum_supervisor_dut_hostname):
+                # continue
+
+                # Get the current isolation status of the port
+                cmd = "sonic-db-cli -n {} CONFIG_DB hget 'FABRIC_PORT|Fabric{}' isolateStatus".format(asicName,
                                                                                                       localPort)
-            cmd_output = duthost.shell(cmd, module_ignore_errors=True)["stdout"].split("\n")
-            tokens = cmd_output[0].split()
-            originalIsolateStatus = tokens[0]
-            pytest_assert(
-                  len(tokens) > 0,
-                  "FABRIC_PORT_TABLE:Fabric{} isolateStatus not found in APPL_DB, {} ".format(localPort, asicName))
-            isolateStatus = tokens[0]
-            pytest_assert(
-                  isolateStatus == "True",
-                  "Port {} APPL_DB initial isolateStatus is '{}', expected False".format(localPort, isolateStatus))
-
-            # If the port was originally not isolsated then restore it
-            if originalIsolateStatus == "False":
-                cmd = "sudo config fabric port unisolate {} -n {}".format(localPort, asicName)
-                cmd_output = duthost.shell(cmd, module_ignore_errors=True)
-                stderr_output = cmd_output["stderr"]
+                cmd_output = duthost.shell(cmd, module_ignore_errors=True)["stdout"].split("\n")
+                tokens = cmd_output[0].split()
+                originalIsolateStatus = tokens[0]
                 pytest_assert(
-                      len(stderr_output) <= 0,
-                      "command: {} failed, error: {}".format(cmd, stderr_output))
+                    originalIsolateStatus == "True" or originalIsolateStatus == "False",
+                    "Port {} CONFIG_DB initial isolateStatus is True, expected False".format(localPort))
+                logger.debug("originalIsolateStatus: {}".format(originalIsolateStatus))
+                try:
+                    # If the port is isolated then temporarily unisolate it
+                    if originalIsolateStatus == "True":
+                        cmd = "sudo config fabric port unisolate {} -n {}".format(localPort, asicName)
+                        cmd_output = duthost.shell(cmd, module_ignore_errors=True)
+                        stderr_output = cmd_output["stderr"]
+                        pytest_assert(
+                            len(stderr_output) <= 0,
+                            "command: {} failed, error: {}".format(cmd, stderr_output))
+
+                    # Check the isolateStatus in CONFIG_DB
+                    cmd = "sonic-db-cli -n {} CONFIG_DB hget 'FABRIC_PORT|Fabric{}' isolateStatus".format(asicName,
+                                                                                                          localPort)
+                    cmd_output = duthost.shell(cmd, module_ignore_errors=True)["stdout"].split("\n")
+                    tokens = cmd_output[0].split()
+                    isolateStatus = tokens[0]
```

Comment:
I agree with the change to not change `originalIsolateStatus`. So removing the original line makes sense. However, `isolateStatus` should be set the pytest_assert() check.